hw/ip/otbn/dv/rig/rig/gens/bad_deep_loop.py

```diff
@@ -0,0 +1,235 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import random
+from typing import Optional, Tuple
+
+from shared.insn_yaml import Insn, InsnsFile
+from shared.operand import ImmOperandType, RegOperandType, OperandType
+
+from .jump import Jump
+from .loop import Loop
+from ..config import Config
+from ..program import ProgInsn, Program
+from ..model import Model
+from ..snippet import LoopSnippet, ProgSnippet, Snippet
+from ..snippet_gen import GenCont, GenRet, SnippetGen
+
+
+class BadDeepLoop(SnippetGen):
+    '''A snippet generator that generates loops to overflow the stack'''
+
+    ends_program = True
+
+    def __init__(self, cfg: Config, insns_file: InsnsFile) -> None:
+        super().__init__()
+        self.jump_gen = Jump(cfg, insns_file)
+        self.loop_gen = Loop(cfg, insns_file)
+
+    def pick_weight(self,
+                    model: Model,
+                    program: Program) -> float:
+        # Only try this if we've got a reasonable amount of room.
+        room = min(model.fuel, program.space)
```

Comment:
nice, a thesaurus line: three different words for roughly the same thing :-)

Reply:
:-) I really struggled to come up with good names! The "fuel" version is "how many more instructions should OTBN execute" vs. "space", which is "how much more space is there in IMEM". I nicked the "fuel" name from Coq, where people use it as a trick when they don't want to (or can't) prove that a definition terminates.